app/mm-redux/utils/post_utils.ts

```diff
@@ -114,7 +116,7 @@ export function canEditPost(state: GlobalState, config: any, license: any, teamI
         }
     } else {
         // Backwards compatibility with pre-advanced permissions config settings.
-        canEdit = isOwner && config.AllowEditPost !== 'never';
+        canEdit = isOwner && (config.AllowEditPost !== 'never' && !isMinimumServerVersion(serverVersion, 6));
```

Comment:
I don't think we need this because hasNewPermissions already checks for a minimum server version of 4.9.0